### PR TITLE
fix controller manager arguments for v1.12-

### PR DESCRIPTION
**Pull request by SataQiu**
**What this PR does / why we need it**:

/kind bug

When I use the command "kubeadm init --kubernetes-version=v1.11.2" to create a k8s cluster, the kube controller manager can not work well beacuse of the bad boot arguments.

The error of kube controller manager pod like this : 
```
Error: unknown flag: --authentication-kubeconfig
Usage:
  kube-controller-manager [flags]
......
```
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #68537

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/controlplane/manifests.go
Comment (rosti) [inline cmd/kubeadm/app/phases/controlplane/manifests.go]: s/miniVersion/v112args

**Review by rosti — CHANGES_REQUESTED** [review]
In its current form this will break the tests in `manifests_test.go`. Those should also be updated with test cases for 1.11.

However, kubeadm 1.12 should only be used to deploy Kubernetes 1.12. It should not be used to deploy any different versions than that. ~So I am reluctant on letting this code in.~ Perhaps @timothysc and @fabriziopandini can have the final word on this.

EDIT: We may need this to keep the e2e tests going.

**Comment by rosti** [comment]
/ok-to-test

Let's see if this fixes kubernetes/kubeadm#1109

**Comment by rosti** [comment]
@SataQiu the failing bazel test is the `manifests_test.go`. Can you fix this, please?

**Comment by neolit123** [comment]
@SataQiu thanks for the PR!
we are going to talk about this in the kubeadm meeting.
something broke one of our e2e tests (not the tests in this PR) and this might be a possible fix.

/hold

**Comment by neolit123** [comment]
@SataQiu are you sure your kube-controller-manager is v1.11.2?

**Comment by neolit123** [comment]
the PR is in the right direction, but it needs some cleanup and unit tests like @rosti suggested.
@SataQiu please coordinate with the changes @rosti suggests.

@fabriziopandini @detiber 
ref: https://github.com/kubernetes/kubernetes/pull/68473

i have verified that the 1.11.3 kube-controller-manager lacks the following flags:
```
		"authentication-kubeconfig"
		"authorization-kubeconfig"
		"client-ca-file"
		"requestheader-client-ca-file"
```

when instantiating a cluster with 1.11.x control plane and kubeadm from latest master the kube-controller-manager pod will fail with the missing arguments like the PR suggest.

```
kube-system   kube-controller-manager-luboitvbox   0/1       CrashLoopBackOff  

<snip>

unknown flag: --authentication-kubeconfig
```

File: cmd/kubeadm/app/phases/controlplane/manifests.go
Comment (kad) [inline cmd/kubeadm/app/phases/controlplane/manifests.go]: We need unit test for both, version v1.12.0  and version v1.11.x

File: cmd/kubeadm/app/phases/controlplane/manifests.go
Comment (kad) [reply cmd/kubeadm/app/phases/controlplane/manifests.go]: In current form, this will not allow to deploy Beta/RC versions of 1.12 to be deployed.
Check when those files are added and use something like 'v1.12.0-beta.0'.
Plus, might be instead of hardcoding constant here, it will be better to declare it in `constants/constants.go`

**Review by kad — CHANGES_REQUESTED** [review]

**Review by timothysc — APPROVED** [review]
/approve

File: cmd/kubeadm/app/phases/controlplane/manifests_test.go
Comment (rosti) [inline cmd/kubeadm/app/phases/controlplane/manifests_test.go]: This test case is enough for me, the rest of the `v1.11.3` in TestGetControllerManagerCommand are not necessary (since we already have them for 1.12)

**Review by rosti — APPROVED** [review]
@SataQiu this looks great now! Thank you very much!

One small nit though, you need to squash your commits and this will be good to go from my perspective.

**Comment by rosti** [comment]
@SataQiu somehow the `v1.11.3` test cases were lost, probably during the squash. Can you re-add them?

**Review by rosti — CHANGES_REQUESTED** [review]
@SataQiu thank you very much! Yet you need to run `./hack/update-gofmt.sh` to get the verify test happy too. 
Once this is done, we are good to go!

**Review by rosti — APPROVED** [review]
@fabriziopandini this is looking good from my side. Can you take a look at it?

**Comment by neolit123** [comment]
/lgtm
merging this and monitoring the tests tonight :+1:

**Pull request merged**
